email_handler.py

```diff
@@ -536,6 +536,55 @@ def handle_email_sent_to_ourself(alias, from_addr: str, msg: Message, user):
     )
 
 
+def apply_dmarc_policy(alias: Alias, contact: Contact, msg: Message) -> Optional[str]:
+    spam_result = msg.get_all(headers.SPAMD_RESULT)
+    if not spam_result:
+        return False
+    spam_entries = [entry.strip() for entry in spam_result[-1].split("\n")]
+    for iPos in range(len(spam_entries)):
+        sep = spam_entries[iPos].find("(")
+        if sep > -1:
+            spam_entries[iPos] = spam_entries[iPos][:sep]
+    if "DMARC_POLICY_REJECT" in spam_entries:
+        return status.E519
+    if (
+        "DMARC_POLICY_SOFTFAIL" in spam_entries
```

Comment:
At the beginning I'd leave DMARC_POLICY_SOFTFAIL out of the quarantine check as it if the sender set the dmarc policy to none, they don't want any special dmarc handling applied on their emails (and therefore accept the risk of having phishing with their domain)

Reply:
When sending spoofed gmail emails that's the dmarc result that rspamd generates. For dmarc the authentication failed (either spf or dkim).

Reply:
Yeah unfortunately gmail has a too relaxed policy but this would protect against phishing attack for other popular domains (facebook.com, airbnb.com, etc all have DMARC enforced)

Reply:
In the future we can create a list of known domain that have dmarc "none" policy but that we want to enforce DMARC on.

Reply:
Ok. For now I've just set to quarantine and reject.

Reply:
Behind an enable flag.